features/cellular/framework/AT/ATHandler.h

```diff
@@ -212,7 +213,8 @@ class ATHandler {
 protected:
     void event();
 #ifdef AT_HANDLER_MUTEX
-    PlatformMutex _fileHandleMutex;
+    rtos::Mutex _fileHandleMutex;
+    rtos::ConditionVariable _oobCv;
```

Comment:
The purpose of `_oobCv` is unclear to me, could add here some description. If `_oobCv` is supposed to guard `_event_id` then `event()` and `unlock()` need locks.

Reply:
oobCv notifies oob completion (it's notified when `event_id` becomes -1).

I agree that it is assuming `event_id` is protected by the `_fileHandleMutex`. That is held as `unlock` is entered, so access there is fine.  If `event()` can be called without that held, then yes it should have a `lock`.

Reply:
`event` is actually called from IRQ context, which means mutex is not possible. But it also means it's pretty safe - this is no worse than the previous `_oob_queued` version, at least. Suggest not changing at this point.